### PR TITLE
Generate human-readable name for downloaded offline map imagery

### PR DESCRIPTION
**Pull request by gino-m**
Towards #1715.

Also deletes a few unused method and converts key functions to suspend functions in the process.

### CODE REVIEW

**Comment by gino-m** [comment]
/gcbrun

@shobhitagarwal1612 or @JSunde PTAL?

**Review by shobhitagarwal1612 — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/system/GeocodingManager.kt [hunk 1]
```diff
@@ -35,39 +36,83 @@ class GeocodingManager
...
+        Address::getCountryName,
+        Address::getAdminArea,
+        Address::getSubAdminArea,
+        Address::getLocality
+      )
+    if (nameComponents.size == 1) {
+      nameComponents.add(0, multipleRegionsLabel)
     }
-    val address = addresses[0]
+    return nameComponents.joinToString(", ").ifEmpty { defaultAreaName }
```
[the remark this answers is not in the document]
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/system/GeocodingManager.kt:68]: Great idea, thanks!

**Pull request merged**